src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java

```diff
@@ -246,19 +246,25 @@ void aggregatePeriodicMetrics() {
     /**
      * Helper for metrics uploader. Also used in tests.
      */
+    @SuppressWarnings("PMD.AvoidCatchingThrowable")
     void publishPeriodicMetrics() {
-        if (!isConnected.get()) {
-            logger.atDebug().log("Cannot publish the metrics. MQTT connection interrupted.");
-            return;
+        try {
+            if (!isConnected.get()) {
+                logger.atDebug().log("Cannot publish the metrics. MQTT connection interrupted.");
+                return;
+            }
+            long timestamp = Instant.now().toEpochMilli();
+            long lastPublish = Coerce.toLong(getPeriodicPublishTimeTopic());
+            Map<Long, List<AggregatedNamespaceData>> metricsToPublishMap =
+                    metricsAggregator.getMetricsToPublish(lastPublish, timestamp);
+            getPeriodicPublishTimeTopic().withValue(timestamp);
+            if (metricsToPublishMap != null && metricsToPublishMap.containsKey(timestamp)) {
+                publisher.publish(MetricsPayload.builder().build(), metricsToPublishMap.get(timestamp));
+                logger.atInfo().event("telemetry-metrics-published").log("Telemetry metrics update published.");
+            }
+        } catch (Throwable t) {
+            logger.atWarn().log("Error collecting telemetry. Will retry.", t);
```

Comment:
nit - remove trailing "."

Reply:
error collecting? shouldn't it be error publishing telemetry? or do you need two separate try/catch to distinguish an error with the metricsAggregator vs an error with the publisher?

Do you need the "Will retry"? Does it retry immediately or at the next publish time

Reply:
It will retry in the next publish